setup/setup_all.sh

```diff
@@ -97,41 +97,49 @@ create_terraform_gcs_bucket() {
 # Run terraform to set up all GCP resources. (Setting up GKE by default)
 init_foundation() {
   # Init Terraform
-  cd "$BASE_DIR"/terraform/stages/foundation
+  cd "${BASE_DIR}"/terraform/stages/foundation
   terraform init -reconfigure -backend-config=bucket="${TF_BUCKET_NAME}"
   
   # Enabling GCP services first.
   terraform apply -target=module.project_services -target=module.service_accounts -auto-approve
-  
+
+
+  # Check for App Engine default service account to check if firebase has been initialized
+  FIREBASE_INIT=""
+  APP_ENGINE_SA=${PROJECT_ID}@appspot.gserviceaccount.com
+  APP_ENGINE_SA_EXISTS=$(gcloud iam service-accounts list | grep -c "$APP_ENGINE_SA")
+  if [[ "${APP_ENGINE_SA_EXISTS}" == 0 ]]; then
+    FIREBASE_INIT="-var=\"firebase_init=true\""
+  fi
```

Comment:
I know Firebase is initialized via AppEngine, is this unique enough to tell whether Firestore exists?

Reply:
This has been a feature request to make this idempotent (https://github.com/hashicorp/terraform-provider-google/issues/10351). I added a comment to revisit this workaround in the future.